asoh/models/ecm/simulator.py

```diff
@@ -0,0 +1,107 @@
+from typing import Literal, Tuple, List
+
+from asoh.models.ecm import EquivalentCircuitModel as ECM
+from asoh.models.ecm import (ECMASOH,
+                             ECMTransientVector,
+                             ECMInput,
+                             ECMMeasurement)
+
+
+class ECMSimulator():
+    """
+    Class used to simulate and store information about an ECM
+
+    Args:
+        asoh: Advanced State of Health (A-SOH) of the system. Used to parametrize the dynamics of the system.
+        transient_state: Initial physical transient state of the ECM. If not provided, will be instantiated based on the
+            A-SOH provided assuming with all values initialized to 0.0
+        initial_input: Initial input of the ECM. If not provided, will be instantiated assuming the system starts at
+            time = 0.0 seconds with a current of 0.0 Amps.
+        current_behavior: Determines how to the total current behaves in-between time steps. Can be either 'constant' or
+            'linear'.
+        keep_history: Boolean to determine whether we wish to keep history of the system.
+    """
+
+    def __init__(self,
+                 asoh: ECMASOH,
+                 transient_state: ECMTransientVector = None,
+                 initial_input: ECMInput = None,
+                 current_behavior: Literal['constant', 'linear'] = 'constant',
+                 keep_history: bool = False,
+                 ) -> None:
+        self.has_C0 = asoh.c0 is not None
```

Comment:
Fair point. I think in part because I know knowledge of the dimensionality of the problem would be useful for the UKF, I am storing some info here. However, I agree, it's probably better to store it where it would be needed (after all, this simulator is not going to be used by the Kalman-based filters). Removing these attributes and marking as resolved.